apis/ingress/v1/settings_types.go

```diff
@@ -20,31 +20,106 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// IdentityProvider see https://www.pomerium.com/docs/identity-providers/
+type IdentityProvider struct {
+	// Provider one of accepted providers https://www.pomerium.com/reference/#identity-provider-name
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=auth0;azure;google;okta;onelogin;oidc;ping;github
+	Provider string `json:"provider"`
+	// URL is identity provider url, see https://www.pomerium.com/reference/#identity-provider-url
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Format=uri
+	// +kubebuilder:validation:Pattern=`^https://`
+	URL *string `json:"url"`
+	// Secret refers to a k8s secret containing IdP provider specific parameters
+	// and must contain at least `client_id` and `client_secret` map values,
+	// an optional `service_account` field, mapped to https://www.pomerium.com/reference/#identity-provider-service-account
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:MinLength=1
+	Secret string `json:"secret"`
+	// ServiceAccountFromSecret is a convenience way to build a value for `idp_service_account` from
+	// secret map values, see https://www.pomerium.com/docs/identity-providers/
+	// +optional
+	ServiceAccountFromSecret *string `json:"serviceAccountFromSecret,omitempty"`
+	// RequestParams see https://www.pomerium.com/reference/#identity-provider-request-params
+	// +optional
+	RequestParams map[string]string `json:"requestParams,omitempty"`
+	// RequestParamsSecret is a reference to a secret for additional parameters you'd prefer not to provide in plaintext
+	// +optional
+	RequestParamsSecret *string `json:"requestParamsSecret,omitempty"`
+	// Scopes see https://www.pomerium.com/reference/#identity-provider-scopes
+	// +optional
+	Scopes []string `json:"scopes,omitempty"`
+
+	// Specifies refresh settings
+	// +optional
+	RefreshDirectory *RefreshDirectorySettings `json:"refresh_directory"`
```

Comment:
we seem to be mixing case? Here we use snake case, but above we're using camel case